### ISSUE TITLE
iOS/Mac Catalyst image localization

### ISSUE THREAD

**Issue by davidbritch** [dotnet/maui#16734]
### Description

The technique for localizing images on iOS are well known. It involves creating folders with the locale name, followed by .lproj, with the images in these folders. These would normally go in a Resources folder. So for MAUI I translated this to the Platforms\iOS\Resources folder:

![image](https://github.com/dotnet/maui/assets/8092460/b9130135-08fc-4331-a75c-9ec52e5a51b6)

The images have a build action of BundleResource.

This doesn't work for iOS, or on Mac Catalyst (using the equivalent MacCatalyst folder).

What does work is putting the .lproj folders in the root of the MAUI project:

![image](https://github.com/dotnet/maui/assets/8092460/61dc0910-2a0f-4582-95ad-8fb9f3abec81)

Localized images are then displayed (on both iOS + MacCat). 

While it works it makes me feel dirty having the root of your project contaminated with Apple only resources.

UPDATE: @rolfbjarne has told me that updating the .csproj to include `<IPhoneResourcePrefix>Platforms/iOS/Resources</IPhoneResourcePrefix>` makes localised images, contained in that specific folder, work on iOS + MacCat.

### Steps to Reproduce

1. Run attached sample on iOS.

### Link to public reproduction project repository

https://github.com/davidbritch/dotnet-maui-issues/tree/main/LocalizationDemo

### Version with bug

7.0.92

### Is this a regression from previous behavior?

Don't think this has ever worked!

### Affected platforms

iOS, macOS

### Did you find any workaround?

Place .lproj folders in the root of your MAUI project.

OR (better)

Update your .csproj with the following:

```xml
	<PropertyGroup Condition="$(TargetFramework.Contains('-ios'))">
		<IPhoneResourcePrefix>Platforms/iOS/Resources</IPhoneResourcePrefix>
	</PropertyGroup>

	<PropertyGroup Condition="$(TargetFramework.Contains('-maccatalyst'))">
		<IPhoneResourcePrefix>Platforms/MacCatalyst/Resources</IPhoneResourcePrefix>
	</PropertyGroup>
```

**Comment by rolfbjarne** [comment]
MAUI already sets `IPhoneResourcePrefix`, but it happens too late:

https://github.com/dotnet/maui/blob/e36d222ecf830f8e92b09e435db3e0f98e58bd5c/src/Controls/src/Build.Tasks/nuget/buildTransitive/netstandard2.0/Microsoft.Maui.Controls.SingleProject.targets#L46

https://github.com/dotnet/maui/blob/e36d222ecf830f8e92b09e435db3e0f98e58bd5c/src/Controls/src/Build.Tasks/nuget/buildTransitive/netstandard2.0/Microsoft.Maui.Controls.SingleProject.targets#L52

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.8.0 Preview 1.0. Repro on iOS 16.4 and Windows 11 .NET 8, not repro on Android 13.0-API33 with below Project:
[LocalizationDemo.zip](https://github.com/dotnet/maui/files/12416563/LocalizationDemo.zip)

iOS:
![image](https://github.com/dotnet/maui/assets/97647558/d58c424c-cd21-4ec8-be24-42e0959b1427)
Android:
![image](https://github.com/dotnet/maui/assets/97647558/b6abd98c-9c4c-472e-9198-f1355a8f2004)

**Comment by ArchimedesFTW** [comment]
@davidbritch does localization of `Info.plist` work for you with your proposed workaround?

I want to ask for location permission with localized text.

I moved `fr.lproj` to the project root to get the flag visible on iOS. I added `InfoPlist.strings` in that folder but it won't show the different text. Instead it shows the text that I added in the `Info.plist`.

![image](https://github.com/dotnet/maui/assets/55234938/9ee3da7e-c562-44f4-a5fe-65195920a8f7)

Any suggestions?

**Comment by davidbritch** [comment]
@ArchimedesFTW It worked for me at the time of writing. But there've been multiple releases since then, and I've not tried it since.

**Comment by janusw** [comment]
> @davidbritch does localization of `Info.plist` work for you with your proposed workaround?
> 
> I want to ask for location permission with localized text.

FYI, I got the permission localization working by placing the lproj folders here (with MAUI 8.0.14):

```
	<ItemGroup Condition="'$(TargetFramework)'=='net8.0-ios'">
		<BundleResource Include="Resources\de.lproj\InfoPlist.strings" />
	</ItemGroup>
```

**Comment by janusw** [comment]
> FYI, I got the permission localization working by placing the lproj folders here (with MAUI 8.0.14):
> 
> ```
> 	<ItemGroup Condition="'$(TargetFramework)'=='net8.0-ios'">
> 		<BundleResource Include="Resources\de.lproj\InfoPlist.strings" />
> 	</ItemGroup>
> ```

Note that this location is slightly different from the one in the workaround mentioned in the original report (i.e. I'm placing the lproj folders into `Resources`, and not directly into the root folder).

Also, I can confirm that the second workaround works well with permission localization: If the lproj folders are in `Platforms\iOS\Resources`, you need the additional

```<IPhoneResourcePrefix>Platforms/iOS/Resources</IPhoneResourcePrefix>```

in your csproj.

**Comment by tdamir** [comment]
In my case a working solution is to move files from ```Platforms/iOS/Resources``` to ```root/Resources``` folder OR move them to Resources/raw but then there is no need for ```<BundleResource>``` include.

Keeping the files in Platforms/iOS/Resources and adding the ```<IPhoneResourcePrefix>Platforms/iOS/Resources</IPhoneResourcePrefix>``` did't work for me. When I tried to publish/archive the project, it complained about not finding a generated Info.Plist file. Also, I have noticed that there were some warnings/errors like "The given path's format is not supported" or something like that while building the project. 

I'm using Visual Studio Version 17.10.0 Preview 3.0. and MAUI 8.0.14.
